### PR TITLE
feat(947): Implement user default collection

### PR DESCRIPTION
**Pull request by Code-Beast**
## Context
Would be nice if users can find all pipeline create by themselves in a default collection.

## Objective
This PR creates a default collection ("My Pipelines") where all pipelines a user creates will be stored when the user create the first pipeline and prevents the creation or renaming of default collection using API. (This PR comes with a change in the schema of collection => new field: type).

## References
Related to [947](https://github.com/screwdriver-cd/screwdriver/issues/947)

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/collections/update.js [hunk 1]
```diff
@@ -22,6 +22,11 @@ module.exports = () => ({
             }
         },
         handler: (request, reply) => {
+            // Check if the collection to be updated has a type 'default'
+            if (request.payload.type === 'default') {
+                throw boom.forbidden('Collection can not be updated to be a default type');
```
Comment (tkyi) [inline plugins/collections/update.js]:
```suggestion
                throw boom.forbidden('Collection can not be updated to type "default"');
```

File: plugins/pipelines/create.js [hunk 2]
```diff
@@ -69,22 +69,71 @@ module.exports = () => ({
...
+                                    // Check if a default pipeline for current user exists
+                                    // create a default collection if the default collection does not exist
+                                    if (!collection) {
+                                        // TODO: get Ids of all pipelines created by current user ??
+                                        // create a default collection
+                                        return collectionFactory.create({
+                                            userId: user.id,
+                                            name: 'My Pipelines',
+                                            description:
+                                                `The default collection of ${user.username}`,
```
Comment (tkyi) [inline plugins/pipelines/create.js]:
```suggestion
                                                `The default collection for ${user.username}`,
```

File: plugins/collections/create.js [hunk 3]
```diff
@@ -21,6 +21,19 @@ module.exports = () => ({
...
+            // Check if the collection to be created has a type 'default'
+            if (request.payload.type === 'default') {
+                return reply(boom.forbidden(
+                    'Collection with type "default" cannot be created by user'
+                ));
+            }
+
+            // If the request doesn't specify the collection type
+            // a 'normal' type is added to the payload
+            if (!request.payload.type) {
```
Comment (adong) [inline plugins/collections/create.js]: if request.payload.type is either undefined or not part of allowed types, then default it to `normal`

File: plugins/collections/get.js [hunk 4]
```diff
@@ -162,9 +184,9 @@ module.exports = () => ({
                 })
                 .catch(err => reply(boom.boomify(err)));
         },
-        response: {
-            schema: getSchema
-        },
+        // response: {
```
Comment (adong) [inline plugins/collections/get.js]: 🤔

**Comment by adong** [comment]
Similar here, `report.20190812.093454.61280.0.001.json`. 
I think thats why `git add -A` is a bad idea.

**Comment by Code-Beast** [comment]
> Add it to ignore file, and dont commit `mw-data/storage.db`

Removed

File: plugins/pipelines/create.js [hunk 5]
```diff
@@ -69,22 +69,69 @@ module.exports = () => ({
...
-                        pipeline.addWebhook(`${request.server.info.uri}/v4/webhooks`)
-                    ]).then((results) => {
-                        const location = urlLib.format({
-                            host: request.headers.host,
-                            port: request.headers.port,
-                            protocol: request.server.info.protocol,
-                            pathname: `${request.path}/${pipeline.id}`
-                        });
+                        })
+                        // TODO: Add the pipeline to the user's My Pipelines collection
```
Comment (tkyi) [inline plugins/pipelines/create.js]: can remove this TODO

**Pull request closed without merging**